### ISSUE TITLE
UI aspects of cross-/multi-user operation

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-ui#635]
How does the UI provide access to other users' workflows?

See also https://github.com/cylc/cylc-uiserver/issues/171

**Comment by hjoliver** [comment]
Currently:
- owner-only access (hub spawns UIS as the authenticated user)

Next step:
- access to a specific other-user (hub spawns UIS as the target user, which authorizes use by the authenticated user)

End-game?
- list of possible or allowed other users?
- gscan show multiple users at once?

**Comment by hjoliver** [comment]
See related comment from @jarich on gscan: https://github.com/cylc/cylc-ui/issues/615#issuecomment-809884828

**Comment by oliver-sanders** [comment]
For documentation, see [cylc-doc/sharing-access-to-workflows](https://cylc.github.io/cylc-doc/stable/html/user-guide/sharing-access-to-workflows.html). This page will shortly be re-written and will likely be moved into an "administrator guide" in the future.

I have chipped off the gscan displaying multiple users workflows into a new issue here - https://github.com/cylc/cylc-ui/issues/1395

And the listing of possible or allowed users here - #1394 

Listing users might not always be possible, if it is it might not always be authorised (Jupyter Hub has an interface for listing users but it is blocked by default) and sites might not want to expose this information either way so we shouldn't rely on the ability to list all users.

What I've suggested in #1394 is that we use a textbox for the server owner and make that box editable as a means to switch server. We can then remember (ideally successful) values as a means to list users. This is probably good enough as I would guess that most users will only ever want to connect to small subset of other users servers. If not we can look at pre-populating this list from another source. The UI Server authentication configuration has previously been suggested as a possible source.

I think that covers everything mentioned in this issue. For anything else multi-user see this GH project: https://github.com/orgs/cylc/projects/3
